session/sql_migration_test.go

```diff
@@ -0,0 +1,776 @@
+					time.Unix(1000, 0), "",
+				)
+				require.NoError(t, err)
+
+				_, err = store.NewSession(
+					ctx, "session3", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "",
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "one session with one privacy flag",
+			populateDB: func(t *testing.T, store *BoltStore,
+				_ accounts.Store) {
+
+				_, err := store.NewSession(
+					ctx, "test", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "",
+					WithPrivacy(PrivacyFlags{ClearPubkeys}),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "one session with multiple privacy flags",
+			populateDB: func(t *testing.T, store *BoltStore,
+				_ accounts.Store) {
+
+				_, err := store.NewSession(
+					ctx, "test", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "",
+					WithPrivacy(PrivacyFlags{
+						ClearChanInitiator, ClearHTLCs,
+						ClearClosingTxIds,
+					}),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "one session with a feature config",
+			populateDB: func(t *testing.T, store *BoltStore,
+				_ accounts.Store) {
+
+				featureConfig := map[string][]byte{
+					"AutoFees":      {1, 2, 3, 4},
+					"AutoSomething": {4, 3, 4, 5, 6, 6},
+				}
+
+				_, err := store.NewSession(
+					ctx, "test", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "",
+					WithFeatureConfig(featureConfig),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "one session with dev server",
+			populateDB: func(t *testing.T, store *BoltStore,
+				_ accounts.Store) {
+
+				_, err := store.NewSession(
+					ctx, "test", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "",
+					WithDevServer(),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "one session with macaroon recipe",
+			populateDB: func(t *testing.T, store *BoltStore,
+				_ accounts.Store) {
+
+				// this test uses caveats & perms from the
+				// tlv_test.go
+				_, err := store.NewSession(
+					ctx, "test", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "foo.bar.baz:1234",
+					WithMacaroonRecipe(caveats, perms),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "one session with macaroon recipe nil caveats",
+			populateDB: func(t *testing.T, store *BoltStore,
+				_ accounts.Store) {
+
+				// this test uses perms from the tlv_test.go
+				_, err := store.NewSession(
+					ctx, "test", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "foo.bar.baz:1234",
+					WithMacaroonRecipe(nil, perms),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "one session with macaroon recipe nil perms",
+			populateDB: func(t *testing.T, store *BoltStore,
+				_ accounts.Store) {
+
+				// this test uses caveats from the tlv_test.go
+				_, err := store.NewSession(
+					ctx, "test", TypeMacaroonAdmin,
+					time.Unix(1000, 0), "foo.bar.baz:1234",
+					WithMacaroonRecipe(caveats, nil),
+				)
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "macaroon recipe with nil perms and caveats",
```

Comment:
Note that this test is new, as this can actually happen for real world sessions.